elastic/eui
# [EuiDataGrid][BugFix]: Fix visible overflowing truncated text

## Summary

closes #7780 

This PR aims to fix an issue on `EuiDataGrid` cells where truncation is applied with `line-clamp` when the following props are set:

```ts
rowHeightsOptions: {
  defaultHeight: {
     lineCount: 1,
  }
}
```

This results in the overflown text being still visible partially. This depends on available space created by the `fontSize` and `cellPadding` passed via `gridStyle`.

_before_
<img width="590" alt="Screenshot 2024-05-28 at 15 06 58" src="https://github.com/elastic/eui/assets/44670957/d3391829-a2f9-4929-a90d-bec29954c587">

To ensure the correct space is considered by `line-clamp`, we need to separate the styles for padding from the truncation step while still ensuring that the padding is included in the height calculation.

_after_
<img width="416" alt="Screenshot 2024-05-28 at 15 23 20" src="https://github.com/elastic/eui/assets/44670957/67e39248-d9d5-4430-beed-4220b88e7ca2">

This PR also adds a more specific story for row height checks: https://eui.elastic.co/pr_7793/storybook/index.html?path=/story/tabular-content-euidatagrid--row-height


## QA

- [ ] open storybook on [production](https://eui.elastic.co/storybook/index.html?path=/story/tabular-content-euidatagrid--playground) and on [this PR branch](https://eui.elastic.co/pr_7793/storybook/index.html?path=/story/tabular-content-euidatagrid--playground) and set the following props:
    - change `fontSize` on `gridStyles` to `"s"`
    ```
    {
      "fontSize": "s",
      "cellPadding": "l",
      "border": "all",
      "stripes": false,
      "header": "shade",
      "footer": "overline",
      "stickyFooter": true,
      "rowHover": "highlight",
      "rowClasses": {}
    }
    ```
    - change `defaultHeight` on `rowHeightsOptions` to `{"lineCount": 1}`
    ```
    {
      "defaultHeight": {
        "lineCount": 1
      },
      "rowHeights": {}
    }
    ```
    - verify that the on the fixed branch storybook the truncated text after the ellipsis is not visible
    - verify no other visual changes have been introduced 


### General checklist

- Browser QA
- Docs site QA -N/A
- Code quality checklist
    - [x] Added or updated VRT images
- Release checklist
- Designer checklist~
